src/workerd/api/util.c++

```diff
@@ -284,4 +285,88 @@ double dateNow() {
   return 0.0;
 }
 
+kj::Maybe<jsg::V8Ref<v8::Object>> cloneRequestCf(
+    jsg::Lock& js,
+    kj::Maybe<jsg::V8Ref<v8::Object>> maybeCf) {
+  KJ_IF_MAYBE(cf, maybeCf) {
+    // In case the cf object has a logging proxy, we want to make sure
+    // the logging is not triggered here when the object is cloned.
+    NoRequestCfProxyLoggingScope noLoggingScope;
+    auto cloned = cf->deepClone(js);
+    maybeWrapBotManagement(js.v8Isolate, cloned.getHandle(js));
+    return kj::mv(cloned);
+  }
+  return nullptr;
+}
+
+void maybeWrapBotManagement(v8::Isolate* isolate, v8::Local<v8::Object> handle) {
+  auto context = isolate->GetCurrentContext();
+  auto botManagement = jsg::v8StrIntern(isolate, "botManagement");
+  v8::Local<v8::Value> maybeBotManagement = jsg::check(handle->Get(context, botManagement));
+  // If the botManagement field exists, we replace its value here with a Proxy object that
+  // will log the first time any of its properties are accessed. Logging will occur only
+  // once per worker instance.
+  //
+  // Replacing the value with a proxy rather than setting an accessor for botManagement on
+  // the request.cf object itself avoids false positives when someone is simply iterating
+  // over the fields of request.cf without actually using them. It also allows us to avoid
+  // having to create accessors or a class to intercept every individual property on the
+  // botManagement object.
+  if (maybeBotManagement->IsObject() && !maybeBotManagement->IsProxy()) {
+    v8::Local<v8::Object> bmObj = maybeBotManagement.As<v8::Object>();
+
+    auto subscribed = jsg::v8StrIntern(isolate, "subscribed");
+
+    // We only want to wrap if the zone is not actually subscribed so that we only get
+    // logging when no subscription is active. We don't, however, want to pass the
+    // subscribed metadata on to the user so we remove it. Unfortunately that means
```

Comment:
I was imagining we'd let the user see the `subscribed` property, it's not really a secret.

Will removing it lead to incorrect behavior when a request is passed through multiple workers in a pipeline? Will the field be lost in the first worker causing subsequent workers to treat the request as if the field wasn't there?

Reply:
Removing it was at the request of the bot management team. But they also want to know if we can just always log whether folks are subscribed or not

Reply:
I see. I think removing it won't work in the case of pipelines. If they'd prefer that we always log anyway, maybe we should go back to that. Sorry for the run-around.